war/pom.xml

```diff
@@ -61,12 +61,8 @@ THE SOFTWARE.
           jars that are not needed in war. most of the exclusions should happen in the core, to make IDEs happy, not here.
         -->
         <exclusion>
-          <groupId>javax.servlet</groupId>
-          <artifactId>servlet-api</artifactId>
```

Comment:
Do you not need to exclude `javax.servlet:javax.servlet-api`?

Reply:
Nope... `<version>[0]</version>` globally excludes it for us!

Reply:
Doing some nonobvious tricks here with dependencies. Have you tried building a core snapshot and then selecting that as a `jenkins.version` from some plugin’s POM to verify that you get the right `mvn dependency:tree` and that `mvn hpi:run` works as expected and so on?

Reply:
> Have you tried building a core snapshot and then selecting that as a jenkins.version from some plugin’s POM to verify that you get the right mvn dependency:tree and that mvn hpi:run works as expected and so on?

No but I have been doing that trick on an internal plugin so that it gets the correct servlet API.

I can give it a go..

Reply:
in `jenkinsci/workflow-step-api-plugin.git`  (sha 8856adca44c476938eb18d62d5a8a5986f60334c)

```
mvn -Denforcer.skip -Djenkins.version=2.80-20171025.175859-1 -Djenkins-war.version=2.80-20171025.180130-1 clean install
mvn -Denforcer.skip -Djenkins.version=2.80-20171025.175859-1 -Djenkins-war.version=2.80-20171025.180130-1 hpi:run
```

install passes all tests and builds ok
hpi:run Jenkins starts up plugin is installed, created a pipeline job with `echo "hello"` build it verified the console output of the run.

enforcer was skipped as project was configured for java7 and obviously has bytecode for java8.
using java.level=8 was not OK as there where syntax errors in some of the generic bounds matching that made it past the java7 compiler